process_qsos_dr7.m

```diff
@@ -0,0 +1,476 @@
+        log10(min(this_unmasked_fine_wavelengths)) - pixel_spacing/2,...
+        width)';...
+        this_unmasked_fine_wavelengths';...
+        logspace(log10(max(this_unmasked_fine_wavelengths)) + pixel_spacing/2,...
+        log10(max(this_unmasked_fine_wavelengths)) + width * pixel_spacing/2,...
+        width)'...
+        ];
+    
+    % [mask_ind] to retain only unmasked pixels from computed absorption profile
+    % this has to be done by using the unmasked_ind which has not yet
+    % been applied this_pixel_mask.
+    ind = (~this_pixel_mask(unmasked_ind));
+    
+    % compute probabilities under DLA model for each of the sampled
+    % (normalized offset, log(N HI)) pairs
+    lenW_unmasked = length(this_unmasked_wavelengths);
+    parfor i = 1:num_C4_samples
+        
+        % Limitting red-shift in the samples
+        this_z_c4 = (this_wavelengths / 1549) - 1;
+        % % absorption corresponding to this sample with one absorption line as a noise model 
+
+                
+        % absorptionL1 = absorptionL1(ind);
+        % c4_muL1     = this_mu     .* absorptionL1;
+        % c4_ML1      = this_M      .* absorptionL1;
+        % %     dla_omega2 = this_omega2 .* absorption.^2;
+        
+        % sample_log_likelihoods_c4L1(quasar_ind, i) = ...
+        %     log_mvnpdf_low_rank(this_flux, c4_muL1, c4_ML1, ...
+        %     this_noise_variance);
+
+        % absorption corresponding to this sample with two absorption lines as a doublet 
+        % compute fine absorption 
+        num_lines=2;
+        if(Rsl==1)
+
+            num_lines = 1;
+            absorptionL1_fine = voigt(padded_fine_wavelengths, sample_z_c4(i), ...
+            nciv_samples(i)*f_L1_nciv ,num_lines , sample_sigma_c4(i)*f_L1_sigma, fine_sigma_pixel);
+            num_lines = 2; 
+            absorptionL2_fine = voigt(padded_fine_wavelengths, sample_z_c4(i), ...
+            nciv_samples(i),num_lines , sample_sigma_c4(i), fine_sigma_pixel);
+
+        else
+            num_lines=1;
+            absorptionL1_fine = voigt0(padded_fine_wavelengths, sample_z_c4(i), ...
+            nciv_samples(i)*f_L1_nciv ,num_lines , sample_sigma_c4(i)*f_L1_sigma);
+            
+            num_lines =2;
+            absorptionL2_fine = voigt0(padded_fine_wavelengths, sample_z_c4(i), ...
+            nciv_samples(i),num_lines , sample_sigma_c4(i));
+        end
+        
+        % average fine absorption and shrink it to the size of original array
+        % as large as the unmasked_wavelengths
+        
+        if SnglMdl==1
+            % For single line model
+            
+            absorptionL1 = zeros(lenW_unmasked, 1);
+            % first and last element of the averaged array are the same as the initial array
+            absorptionL1(1,1) = absorptionL1_fine(1); 
+            absorptionL1(end,1) = absorptionL1_fine(end);
+            
+            % Averaging
```

Comment:
I think I need more explanations to understand this averaging doing.

Reply:
I will put some clear comments and write a separate function for that.